ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -142,13 +152,63 @@ private String injectCss(@NonNull final String content) {
         @Nullable
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
-            if (url.endsWith(".jpg")) {
-                //TODO: Load local Image WebResource
-            } else if (url.endsWith(".css")) {
-                //TODO: Load CSS WebResource
+            String extension = MimeTypeMap.getFileExtensionFromUrl(url);
+
+            WebResourceResponse response = null;
+
+            String mimeType;
+
+            switch (extension) {
+                case "jpg":
+                case "png":
+                case "bmp":
+                case "gif":
+                    mimeType = String.format("image/%s", "jpg".equals(extension) ? "jpeg" : extension);
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+                case "css":
+                    mimeType = "text/css";
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+                default:
+                    mimeType = "application/octet-stream";
+                    response = getResponseFromFile(mimeType, url);
+            }
+
+            if (response == null) {
+                return returnWebResponse(mimeType, new ByteArrayInputStream("".getBytes()));
```

Comment:
what I meant about returning a Not Found response is return a 404 Response, not just an empty resource

Reply:
hmm, the response code part of the API is only available starting in Lollipop. From Lollipop on we should return a 404. otherwise the data from the web response object should probably just be null (assuming it doesn't cause issues on older versions of android)